src/c_api/c_api_ndarray.cc

```diff
@@ -267,13 +275,13 @@ int MXImperativeInvoke(AtomicSymbolCreator creator,
 
   if (outarray == nullptr) {
     ret->ret_handles.clear();
-    for (int i = 0; i < infered_num_outputs; ++i) {
+    for (int i = 0; i < num_visible_outputs; ++i) {
       ret->ret_handles.push_back(
         reinterpret_cast<NDArrayHandle>(new NDArray(std::move(ndoutputs[i]))));
     }
     *outputs = dmlc::BeginPtr(ret->ret_handles);
   } else {
-    for (int i = 0; i < infered_num_outputs; ++i) {
+    for (int i = 0; i < *num_outputs; ++i) {
       *outarray[i] = std::move(ndoutputs[i]);
     }
   }
```

Comment:
I've add this file to make the check pass. Will rebase after that PR gets merged.